scripts/setup/nxp/update_nxp_sdk.py

```diff
@@ -0,0 +1,98 @@
+#!/usr/bin/env python3
+#
+# Copyright (c) 2024 Project CHIP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import argparse
+import os
+import subprocess
+import sys
+
+CHIP_ROOT = os.path.abspath(os.path.join(os.path.dirname(__file__), '../../..'))
+
+# Array listing all SDK and default targeted path for NXP SDK storage
+ALL_PLATFORM_SDK = [
+    {'plat_name': 'k32w0', 'plat_sdk_manifest_path': 'third_party/nxp/nxp_matter_support/github_sdk/k32w0'},
```

Comment:
Consider replacing stringly-typed data with something like `@dataclass` or similar. 

a `[p.name for p in ALL_PLATFORM_SDK]` is more readable than the existing `list(map(lambda p: p['plat_name'], ALL_PLATFORM_SDK))`

Reply:
I did an update in the last commit to use dataclass. Thanks for your recommendation.